src/components/PublishingStatus.tsx

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import { Dropdown } from "react-bootstrap";
+import { useData } from "../DataContext";
+
+/* eslint-disable react/no-array-index-key */
+export function PublishingStatus(): JSX.Element | null {
+  const { publishResults } = useData();
+  if (publishResults.size === 0) {
+    return null;
+  }
+  const getStatusCount = (status: Array<PublishStatus>, type: string): number =>
+    status.filter((s) => s.status === type).length;
+  const getLastRejectedReason = (
+    status: Array<PublishStatus>
+  ): string | undefined => {
+    const lastRejected = status
+      .slice()
+      .reverse()
+      .find((s) => s.status === "rejected");
+    return lastRejected ? lastRejected.reason : undefined;
+  };
+
+  return (
+    <Dropdown>
+      <Dropdown.Toggle
+        as="button"
+        id="publishing-status-dropdown"
+        key="publishing-status-dropdown"
+        className="btn"
+        style={{ paddingTop: "6px", paddingBottom: "4px" }}
+        aria-label="publishing status"
+        tabIndex={0}
+      >
+        <span className="simple-icon-share" />
+      </Dropdown.Toggle>
+      <Dropdown.Menu>
+        <Dropdown.Item key="publishing-status-header">
+          <div className="bold">
+            <h3>Publishing Status</h3>
+          </div>
+        </Dropdown.Item>
+        {publishResults
+          .map((status, relayUrl) => {
+            const numberFulfilled = getStatusCount(status, "fulfilled");
```

Comment:
It would be more readable if the inner part of this .map is it's own component, so you don't need to write so much typescript within this jsx block.